### PR TITLE
vim-gtk 8.2.2650 (new formula)

### PR DESCRIPTION
**Pull request by brewfixvim**
Adds a new vim formula with X11 GUI support to include much needed features in the build e.g. the clipboard utility

### CODE REVIEW

**Comment by iMichka** [comment]
Cool, thanks for working on this.

Just to warn you ahead, we won't be able to merge this PR in this repo right away. You can use it for the CI, but at the very end the pull request needs to be opened against https://github.com/Homebrew/homebrew-core. We are in the process to deprecate the linux repo and only have one single repo: to ease the migration on our end we do not accept any bigger pull requests here.

For the build failures: you need to remove the bottle block. And do not modify the vim.rb formula.

Also, this formula will conflict with vim as-is: this means that you can't install vim and vim-gtk at the same time.
Either we need to add `conflicts_with` statements in both formulae, or make this formula keg_only (which means it will not be linked in our `bin` folder and not put in the PATH.

**Comment by brewfixvim** [comment]
>Submitting a new formula that builds vim with GTK support should be very simple actually. You can just take the existing vim formula, make the change to the configure script and add GTK3 and any other new dependencies, and change the name to gvim to match what other Linux distros do. If you submit a new PR that adds this as a formula I'd be happy to help you get it working.

Yikes! I was led to believe it would be just a [tiny change to the configure script](https://github.com/Homebrew/linuxbrew-core/pull/22774#issuecomment-810557073)

> For the build failures: you need to remove the bottle block. And do not modify the vim.rb formula.
> 
> Also, this formula will conflict with vim as-is: this means that you can't install vim and vim-gtk at the same time.
> Either we need to add `conflicts_with` statements in both formulae, or make this formula keg_only (which means it will not be linked in our `bin` folder and not put in the PATH.

I don't even know what this means.

File: Formula/vim.rb [hunk 1]
```diff
@@ -55,7 +55,6 @@ def install
                           "--enable-rubyinterp",
                           "--enable-python3interp",
                           "--enable-gui=no",
-                          "--without-x",
                           "--enable-luainterp",
```
Comment (iMichka) [inline Formula/vim.rb:58]: Please do not modify the vim.rb file.

**Comment by iMichka** [comment]
> Yikes! I was led to believe it would be just a tiny change to the configure script

Almost, there are just a few details to handle to be able to ship this.

>> Also, this formula will conflict with vim as-is: this means that you can't install vim and vim-gtk at the same time.
Either we need to add conflicts_with statements in both formulae, or make this formula keg_only (which means it will not be linked in our bin folder and not put in the PATH.

> I don't even know what this means.

I'll explain it to you in more details; but I need to gather feedback from the @Homebrew/core maintainers first.
We would like to add a vim formula with gtk support. How should we handle the conflicts between this new formula and the standard formula? Would @Homebrew/core accept this formula?

File: Formula/vim-gtk.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  depends_on "ruby"
+
+  uses_from_macos "ncurses"
+
+  conflicts_with "ex-vi",
+    because: "vim and ex-vi both install bin/ex and bin/view"
+
+  conflicts_with "macvim",
+    because: "vim and macvim both install vi* binaries"
+
```
Comment (SMillerDev) [inline Formula/vim-gtk.rb:24]: It also conflicts with `vim` and the other way around for all of these

File: Formula/vim-gtk.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
+class VimGtk < Formula
+  desc "Vi 'workalike' with many additional features"
+  homepage "https://www.vim.org/"
+  # vim should only be updated every 25 releases on multiples of 25
+  url "https://github.com/vim/vim/archive/v8.2.2650.tar.gz"
+  sha256 "0284b0e66287641b196b29496ea7d44ad5a728da0fc3b8eea36f1ed3636b661e"
+  license "Vim"
+  head "https://github.com/vim/vim.git"
+
+  bottle do
```
[the remark this answers is not in the document]
Comment (brewfixvim) [reply Formula/vim-gtk.rb]: Done

File: Formula/vim.rb (hunk 1 above)
Comment (brewfixvim) [reply Formula/vim.rb]: Done

File: Formula/vim-gtk.rb (hunk 2 above)
Comment (brewfixvim) [reply Formula/vim-gtk.rb:24]: How should we go about this?

**Comment by brewfixvim** [comment]
@danielnachun some help around here?

**Comment by SMillerDev** [comment]
I don't see how more maintainers will help your case here. This needs to be submitted to https://github.com/Homebrew/homebrew-core since it isn't Linux specific. Also keep in mind that everyone here is a volunteer and helps when they feel like it, there is no need to tag people to make it go faster.

**Comment by brewfixvim** [comment]
> I don't see how more maintainers will help your case here. This needs to be submitted to https://github.com/Homebrew/homebrew-core since it isn't Linux specific. Also keep in mind that everyone here is a volunteer and helps when they feel like it, there is no need to tag people to make it go faster.

Should i close this pull request and open a fresh one hombrew-core?

File: Formula/vim-gtk.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
+class VimGtk < Formula
+  desc "Vi 'workalike' with many additional features"
+  homepage "https://www.vim.org/"
+  # vim should only be updated every 25 releases on multiples of 25
```
Comment (Rylan12) [inline Formula/vim-gtk.rb:4]: This needs to be added to the `throttled_formulae.json` list as well.

**Comment by brewfixvim** [comment]
> Since this will need to be submitted to Homebrew/homebrew-core anyway may as well do it now to not waste time

I've opened a new pull request there. Here's the link - https://github.com/Homebrew/homebrew-core/pull/74263

File: Formula/vim-gtk.rb (hunk 4 above)
Comment (brewfixvim) [reply Formula/vim-gtk.rb:4]: What is meant by throttled formulae?

**Pull request closed without merging**